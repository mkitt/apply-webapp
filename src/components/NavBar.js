// @flow
import React from 'react'
import glamorous from 'glamorous'
import { BASE_URL } from '../constants/networking'
import ImageLink from './elements/Link'
import TextLink from './elements/TextLink'
import { ChevronIconThin } from './elements/Icons'

type Props = {
  goBack: Function,
}

export default (props: Props) => (
  <View>
    <Nav>
      <Pulled>
        { props.goBack &&
          <BackButton onClick={props.goBack} role="button">
            <ChevronIconThin />
          </BackButton>
        }
      </Pulled>
      <div>
        <Pushed>
          <User>
            <span><span role="img" aria-label="hi">ðŸ‘‹</span> Jeff Casimir</span>
            <span> / </span>
            <TextLink to="/">Logout</TextLink>
            <span> / </span>
            <TextLink to="/applicant/create">Create Applicant</TextLink>
          </User>
          <ImageLink to="/">
            <img
              alt="turing school mark"
              src={`${BASE_URL}/turing-school-mark-256.png`}
              width={48}
              height={48}
            />
          </ImageLink>
        </Pushed>
      </div>
    </Nav>
  </View>
)

// -------------------------------------

const View = glamorous.div({
  height: 112,
  backgroundColor: '#fff',
})

const Nav = glamorous.nav({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  maxWidth: 1392,
  height: '100%',
  marginLeft: 'auto',
  marginRight: 'auto',
  paddingRight: 16,
  paddingLeft: 16,
})

const Pulled = glamorous.div({
  display: 'flex',
  alignItems: 'center',
})

const Pushed = glamorous.div({
  display: 'flex',
  alignItems: 'center',
})

const User = glamorous.span({
  fontSize: 14,
  color: '#808080',
  marginRight: 40,
})

const BackButton = glamorous.button({
  color: '#999',
})
