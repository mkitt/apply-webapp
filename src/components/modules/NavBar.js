// @flow
import React from 'react'
import glamorous from 'glamorous'
import { COLORS } from '../../constants/theme'
import ImageLink from '../elements/Link'
import TextLink from '../elements/TextLink'
import TuringMark from '../elements/TuringMark'
import { ChevronIconThin } from '../elements/Icons'

type Props = {
  goBack: Function,
}

export default (props: Props) => (
  <Nav>
    <View>
      <Column>
        { props.goBack &&
          <BackButton onClick={props.goBack} role="button">
            <ChevronIconThin />
          </BackButton>
        }
      </Column>
      <Column>
        <User>
          <span><span role="img" aria-label="hi">ðŸ‘‹</span> Jeff Casimir</span>
          <span> / </span>
          <TextLink to="/login">Logout</TextLink>
        </User>
        <ImageLink to="/">
          <TuringMark />
        </ImageLink>
      </Column>
    </View>
  </Nav>
)

// -------------------------------------

export const Nav = glamorous.nav({
  height: 112,
  backgroundColor: COLORS.WHITE,
})

const View = glamorous.div({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  maxWidth: 1392,
  height: '100%',
  marginLeft: 'auto',
  marginRight: 'auto',
  paddingRight: 8,
  paddingLeft: 8,
})

const Column = glamorous.div({
  display: 'flex',
  alignItems: 'center',
  height: '100%',
})

const User = glamorous.span({
  fontSize: 14,
  color: COLORS.GREY_8,
  marginRight: 40,
})

const BackButton = glamorous.button({
  color: COLORS.GREY_9,
})
