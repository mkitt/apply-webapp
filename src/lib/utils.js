// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
