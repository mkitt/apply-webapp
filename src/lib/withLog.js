import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
