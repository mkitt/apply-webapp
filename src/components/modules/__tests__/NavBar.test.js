import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { noop } from '../../../lib/utils'
import NavBar from '../NavBar'

describe('<NavBar />', () => {
  it('shows the correct snapshot tree for the <NavBar> module', () => {
    const component = (
      <MemoryRouter>
        <NavBar
          goBack={noop}
        />
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
