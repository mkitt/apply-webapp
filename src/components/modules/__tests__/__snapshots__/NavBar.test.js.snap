// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> shows the correct snapshot tree for the <NavBar> module 1`] = `
.css-0,
[data-css-0] {
  height: 112px;
  background-color: #fff;
}

.css-1,
[data-css-1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1392px;
  height: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-right: 8px;
  padding-left: 8px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-2,
[data-css-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  height: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-3,
[data-css-3] {
  color: #999;
}

.css-4,
[data-css-4] {
  font-size: 14px;
  color: #808080;
  margin-right: 40px;
}

.css-5,
[data-css-5] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<nav
  className="css-0"
>
  <div
    className="css-1"
  >
    <div
      className="css-2"
    >
      <button
        className="css-3"
        onClick={[Function]}
        role="button"
      >
        <svg
          className="css-nil"
          height={24}
          viewBox="0 0 16 30"
          width={12}
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            transform="translate(-8.000000, -1.000000)"
          >
            <polyline
              points="23 31 9 16.7142331 23 1"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      className="css-2"
    >
      <span
        className="css-4"
      >
        <span>
          <span
            aria-label="hi"
            role="img"
          >
            ðŸ‘‹
          </span>
           Jeff Casimir
        </span>
        <span>
           / 
        </span>
        <a
          className="css-5"
          href="/login"
          onClick={[Function]}
        >
          Logout
        </a>
      </span>
      <a
        className="css-nil"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="turing school mark"
          height={48}
          src="/turing-school-mark-256.png"
          width={48}
        />
      </a>
    </div>
  </div>
</nav>
`;
